### PR TITLE
future.cr -> async

### LINKED ISSUE #7: Migrate from futures
Strip dependency on `future.cr`, potentially replace with `promise.cr`

### PR DESCRIPTION
**Pull request by watzon**
Closes #7

Figured I'd go ahead and PR this and see how you feel about it. As mentioned in #7, my `Async::Future` is a bit more powerful than the old `Future` that use to be in the standard library. I've tried to take inspiration from both Python's asyncio and JavaScript's promises.

### CODE REVIEW

File: shard.yml [hunk 1]
```diff
@@ -9,8 +9,10 @@ dependencies:
   active-model:
     github: spider-gazelle/active-model
     version: ~> 2
-  future:
-    github: crystal-community/future.cr
+  async:
+    # github: protoncr/async
+    # branch: main
+    path: ../../personal/async
```
Comment (caspiano) [inline shard.yml]: Could you change this to a pinned version?

**Comment by caspiano** [comment]
The team that maintains this library has its own asynchronous abstraction of Fibers as [Promise](https://github.com/spider-gazelle/promise). We've been using that library in production for a while now and I currently don't have the bandwidth to review your library. I'll merge once I have had time to dig through your library.

**Comment by watzon** [comment]
Sounds good man, just trying to help close some issues while I also create some new ones

File: shard.yml (hunk 1 above)
Comment (watzon) [reply shard.yml]: Whoops, will do

Comment (watzon) [reply shard.yml]: Done!

**Comment by watzon** [comment]
Tagged it wrong initially so the build failed, but it should pass now

**Pull request closed without merging**